File: questao3/resposta.js

// const faturamento_diario = [
//     1200, 1500, 0, 0, 1800, 1900, 0, 2000, 2100, 0, 0, 1750, 1600,
// ]
// const start = performance.now();
// const faturamentoValido = faturamento_diario.filter(e=> e > 0 )
// const menorFaturamento = Math.min(...faturamentoValido)
// const maiorFaturamento = Math.max(...faturamentoValido)
// const soma = faturamentoValido.reduce((valorAcumulado, valorAtual)=> valorAcumulado + valorAtual, 0) 
// const media = soma / faturamentoValido.length; 
// const diasAcimaDoFaturamento = faturamento_diario.filter(valor => valor > media);
// console.log(`Menor valor de faturamento: R$ ${menorFaturamento}`)
// console.log(`Maior valor de faturamento: R$ ${maiorFaturamento}`)
// console.log(`Número de dias com faturamento acima da média: ${diasAcimaDoFaturamento.length} com os valores ${diasAcimaDoFaturamento}`)
// const end = performance.now();
// const executionTime = (end - start) / 1000; 

// console.log(`Tempo de execução: ${executionTime.toFixed(6)} segundos`);



const fs = require('fs');

const rawData = fs.readFileSync('./questao3/dados.json');
const faturamento_diario = JSON.parse(rawData);
const start = performance.now();
const valores = faturamento_diario.map(entry => entry.valor);
const faturamentoValido = valores.filter(valor => valor > 0);
const menorFaturamento = Math.min(...faturamentoValido);
const maiorFaturamento = Math.max(...faturamentoValido);
const soma = faturamentoValido.reduce((valorAcumulado, valorAtual) => valorAcumulado + valorAtual, 0);
const media = soma / faturamentoValido.length;
const diasAcimaDoFaturamento = valores.filter(valor => valor > media);
console.log(`Menor valor de faturamento: R$ ${menorFaturamento.toFixed(2)}`);
console.log(`Maior valor de faturamento: R$ ${maiorFaturamento.toFixed(2)}`);
console.log(`Número de dias com faturamento acima da média: ${diasAcimaDoFaturamento.length} com os valores ${diasAcimaDoFaturamento}`);
const end = performance.now();
const executionTime = (end - start) / 1000;

console.log(`Tempo de execução: ${executionTime.toFixed(6)} segundos`);

File: questao3/resposta.py
# import time
# faturamentoDiario = [
#     1200, 1500, 0, 0, 1800, 1900, 0, 2000, 2100, 0, 0, 1750, 1600,
# ]
# startTime = time.time()

# faturamentoValido = [valor for valor in faturamentoDiario if valor > 0]

# menorFaturamento = min(faturamentoValido)
# maiorFaturamento = max(faturamentoDiario)
# mediaFaturamento = sum(faturamentoValido) / len(faturamentoValido)
# diasAcimaDaMedia = sum(1 for valor in faturamentoValido if valor > mediaFaturamento)
# endTime = time.time()
# print(f"Menor valor de faturamento: R$ {menorFaturamento}")
# print(f"Maior valor de faturamento: R$ {maiorFaturamento}")
# print(f"Número de dias com faturamento acima da média: {diasAcimaDaMedia}")

# executionTime = startTime - endTime
# print(f"Tempo de execução: {executionTime:.6f} segundos")

import time
import xml.etree.ElementTree as ET

tree = ET.parse('./questao3/dados.xml')
root = tree.getroot()

faturamentoDiario = [float(row.find('valor').text) for row in root.findall('row')]

startTime = time.time()

faturamentoValido = [valor for valor in faturamentoDiario if valor > 0]

menorFaturamento = min(faturamentoValido)
maiorFaturamento = max(faturamentoDiario)
mediaFaturamento = sum(faturamentoValido) / len(faturamentoValido)
diasAcimaDaMedia = sum(1 for valor in faturamentoValido if valor > mediaFaturamento)

endTime = time.time()

print(f"Menor valor de faturamento: R$ {menorFaturamento:.2f}")
print(f"Maior valor de faturamento: R$ {maiorFaturamento:.2f}")
print(f"Número de dias com faturamento acima da média: {diasAcimaDaMedia}")

executionTime = endTime - startTime
print(f"Tempo de execução: {executionTime:.6f} segundos")
